core/src/main/java/org/elasticsearch/cluster/routing/allocation/decider/DiskThresholdDecider.java

```diff
@@ -347,8 +348,8 @@ public Decision canAllocate(ShardRouting shardRouting, RoutingNode node, Routing
         }
 
         // a flag for whether the primary shard has been previously allocated
-        IndexMetaData indexMetaData = allocation.metaData().getIndexSafe(shardRouting.index());
-        boolean primaryHasBeenAllocated = shardRouting.primary() && shardRouting.allocatedPostIndexCreate(indexMetaData);
+        boolean primaryHasBeenAllocated = shardRouting.primary() &&
```

Comment:
I think the original intention here is to see whether this is a new primary with no data, which means we should ignore low watermark checks (it's empty anyway). I don't think we should consider this for snapshot restore. Shrink is trickier - on the one hand we do duplicate the data (we kick off a merge) but on the other hand, all shard copies of the source index are supposedly on this node, so we don't really have a choice.